taskipy/plugin.py

```diff
@@ -0,0 +1,64 @@
+# pylint: disable=import-error
+
+from pathlib import Path
+
+from cleo.commands.command import Command
+from cleo.exceptions import LogicException
+from cleo.helpers import argument, option
+from poetry.console.application import Application
+from poetry.plugins.application_plugin import ApplicationPlugin
+
+from taskipy.cli import run
+from taskipy.task_runner import TaskRunner
+
+
+class TaskipyCommand(Command):
+    name = 'task'
+    description = 'Taskipy - the complementary task runner for python.'
+    options = [option('list', 'l', 'Show list of available tasks.')]
+    arguments = [
+        argument('name', 'Name of the task.', optional=True),
+        argument(
+            'args', 'Arguments to pass to the task.', optional=True, multiple=True
+        ),
+    ]
+
+    def handle(self) -> int:
+        name = self.argument('name')
+        args = self.argument('args')
+        return run(name=name, args=args, list=self.option('list'))
+
+
+class BaseCommand(Command):
+    description = 'A taskipy task'
+    arguments = [
+        argument(
+            'args', 'Arguments to pass to the task.', optional=True, multiple=True
+        ),
+    ]
+    hidden = True
+
+    @classmethod
+    def from_task(cls, task: str) -> 'BaseCommand':
+        return type(f'{task.capitalize()}Class', (cls,), {'name': task})()
+
+    def handle(self) -> int:
+        return run(name=self.name, args=self.argument('args'))
+
+
+class Plugin(ApplicationPlugin):
+    def activate(self, application: Application) -> None:
+        # Register `poetry task`
```

Comment:
If/when this feature is added, what do you all think about defaulting to the namespaced `poetry task` and having an option in `[tool.taskipy.settings]` to override that behavior and expose the tasks on `poetry` instead? This allows the user of the package to choose which they prefer but default to the safer namespaced option.

Reply:
that makes perfect sense to me. perhaps also an environment variable if we want the setting to be enabled not only at the project-level but also separately by individuals